elide-datastore/elide-datastore-hibernate5/src/main/java/com/yahoo/elide/datastores/hibernate5/HibernateTransaction.java

```diff
@@ -207,6 +252,71 @@ public void commit() {
     }
 
     @Override
+    public <T> Collection filterSortOrPaginateCollection(final Collection collection, final Class<T> entityClass,
+                                                         final EntityDictionary dictionary,
+                                                         final Optional<Set<Predicate>> filters,
+                                                         final Optional<Sorting> sorting,
+                                                         final Optional<Pagination> pagination) {
+        if (((collection instanceof AbstractPersistentCollection))
+                && (filters.isPresent() || sorting.isPresent() || pagination.isPresent())) {
+
+            String filterString = "";
+
+            // apply filtering - eg where clause's
+            if (filters.isPresent()) {
+                filterString += hqlFilterOperation.applyAll(filters.get());
+            }
+
+            // add sorting into HQL string query generation
+            if (sorting.isPresent() && !sorting.get().isEmpty()) {
+
+                final Map<String, Sorting.SortOrder> validSortingRules = sorting.get().getValidSortingRules(
+                        entityClass, dictionary
+                );
+                String additionalHQL = "";
+                if (!validSortingRules.isEmpty()) {
+
+                    final List<String> ordering = new ArrayList<>();
+                    // pass over the sorting rules
+                    validSortingRules.entrySet().stream().forEachOrdered(entry ->
+                            ordering.add(entry.getKey() + " " + (entry.getValue().equals(Sorting.SortOrder.desc)
+                                    ? "desc"
+                                    : "asc"))
+                    );
+                    additionalHQL += "order by " + StringUtils.join(ordering, ",");
```

Comment:
I'm wondering if we should do a Builder pattern here.  I would consider repurposing the HqlFilterOperation (maybe renamed to HqlBuilder) add add methods like: (addSorting, addFilter, addPagination).  This would also break up this function into smaller bits.